### DISCUSSION TITLE
Meshcentral & Tailscale issues [Resolved]

### DISCUSSION THREAD

**Discussion by jbob007** [Ylianst/MeshCentral#6628] [General]
**Setup:**
I have MeshCentral (MC) installed on a Proxmox server running in an LXC container (not using Docker).
I used the [MeshCentral installation script](https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/install/meshcentral-install.sh) for easy hands off. Check the link to see the commands used for installation.
Caddy is set up as the reverse proxy, and I'm using Cloudflare for custom domains, all routed through my Tailscale network.
I can access MeshCentral through my custom domain without any issues.
I’ve also successfully downloaded the MeshCentral Agent on a client machine.

**Issue:**
Despite the agent being installed and accessible through my custom domain, it's not showing up in the MeshCentral dashboard.

**What I've Tried:**
The MeshCentral server is accessible via my custom domain through Tailscale, but the agent is not registering.
I’ve verified that the agent is installed correctly, but I don’t see any indication of it on the dashboard.

**Question:**
What do I need to change or check in the configuration to get the MeshCentral Agent to show up in the dashboard? 

I'm new to this so if layman terms/steps could be given that would be great!

**Comment by si458** [comment]
what does your config.json look like?
did you edit it at all when you set it up?
if not its running in LAN mode which means it wont work outside the lxc containers subnet mask (local network)

**Reply by jbob007** [reply]
Here is the modification I did to my config file.

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the>
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "remote.secret.mydomain.com",
    "WANonly": true,
    "_LANonly": true,
    "sessionKey": "MySuperSecretPassword56",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```

**Reply by si458** [reply]
ok so is the IP address of `remote.secret.mydomain.com` resolved as the tailscale IP or the vms ip?
if its the tailscale IP, then i think you would need to make sure all ur remote devices had the tailscale software installed/setup etc and can talk to the meshcentral server

**Reply by jbob007** [reply]
My CaddyFile is set to the lxc container's IP. I have tailscale installed on each endpoint and I can access MC through both the tailscale IP and the custom domain on the remote devices but the installed agent doesn't register in the dashboard. I saw someone mentioned something  on an older post about being able to configure the agent installer to point to something else other than the localhost but there was no details as to how or further clarification on the subject.

**Reply by si458** [reply]
right ok so you are using a reverse proxy - https://ylianst.github.io/MeshCentral/meshcentral/#nginx-reverse-proxy-setup
so you need to add the following, then restart meshcentral, and try again,
```
"settings": {
  "trustedProxy": "ip.of.caddy.server"
},
"domains": {
  "": {
    "certurl": "https://remote.secret.mydomain.com"
  }
```

**Reply by jbob007** [reply]
I've uninstalled and re-downloaded and reinstalled the agent and the issue still persists. The agent installer shows the server url pointing to the correct domain as shown here:
![image](https://github.com/user-attachments/assets/039fcd6c-ca59-46a6-8974-013eae299ff9)
Here's the agent running and connected after the install on the remote device and access to the admin portal:
![image](https://github.com/user-attachments/assets/3446a433-12e5-4142-b729-468ba38a2292)


![image](https://github.com/user-attachments/assets/83ea8425-784f-41f6-ba98-d9734685aca7)

**Reply by jbob007** [reply]
> right ok so you are using a reverse proxy - https://ylianst.github.io/MeshCentral/meshcentral/#nginx-reverse-proxy-setup so you need to add the following, then restart meshcentral, and try again,
> 
> ```
> "settings": {
>   "trustedProxy": "ip.of.caddy.server"
> },
> "domains": {
>   "": {
>     "certurl": "https://remote.secret.mydomain.com"
>   }
> ```

Sorry didn't see the reply before sending my response. Trying it now and will let you know.

**Reply by jbob007** [reply]
It's working now! Thanks!
